$OpenBSD: patch-plugins_mouse_gsd-mouse-manager_c,v 1.1 2009/02/02 15:13:08 ajacoutot Exp $

GNOME bug #562977:
fix crash with X servers that don't provide XInput (e.g. VNC...)

--- plugins/mouse/gsd-mouse-manager.c.orig	Tue Nov 25 00:02:41 2008
+++ plugins/mouse/gsd-mouse-manager.c	Fri Jan 30 22:55:58 2009
@@ -327,9 +327,14 @@ devicepresence_filter (GdkXEvent *xevent,
 static void
 set_devicepresence_handler (GsdMouseManager *manager)
 {
-        Display *display = gdk_x11_get_default_xdisplay ();
+        Display *display;
         XEventClass class_presence;
         int xi_presence;
+
+        if (!supports_xinput_devices ())
+                return;
+
+        display = gdk_x11_get_default_xdisplay ();
 
         gdk_error_trap_push ();
         DevicePresence (display, xi_presence, class_presence);
